common/Windows/SecondaryWindow.cs

```diff
@@ -0,0 +1,328 @@
+            var primaryWindowLeftOffset = PrimaryWindow.AppWindow.Position.X;
+            var primaryWindowTopOffset = PrimaryWindow.AppWindow.Position.Y;
+            var primaryWindowHalfWidth = (PrimaryWindow.Width * dpi) / 2;
+            var primaryWindowHalfHeight = (PrimaryWindow.Height * dpi) / 2;
+
+            // Derive secondary window dimensions
+            var secondaryWindowHalfWidth = (Width * dpi) / 2;
+            var secondaryWindowHalfHeight = (Height * dpi) / 2;
+            var secondaryWindowLeftOffset = primaryWindowLeftOffset + primaryWindowHalfWidth - secondaryWindowHalfWidth;
+            var secondaryWindowTopOffset = primaryWindowTopOffset + primaryWindowHalfHeight - secondaryWindowHalfHeight;
+
+            // Move and resize secondary window
+            this.MoveAndResize(secondaryWindowLeftOffset, secondaryWindowTopOffset, Width, Height);
+        }
+    }
+
+    /// <summary>
+    /// Show secondary window in taskbar.
+    /// </summary>
+    /// <remarks>
+    /// This is specifically required when a window owner is set, where by
+    /// default the secondary window is not visible in the taskbar.
+    /// </remarks>
+    private void ShowInTaskbar()
+    {
+        var hwnd = (HWND)this.GetWindowHandle();
+        var exStyle = PInvoke.GetWindowLong(hwnd, WINDOW_LONG_PTR_INDEX.GWL_EXSTYLE) | (int)WINDOW_EX_STYLE.WS_EX_APPWINDOW;
+        _ = PInvoke.SetWindowLong(hwnd, WINDOW_LONG_PTR_INDEX.GWL_EXSTYLE, exStyle);
+    }
+
+    private static void SetWindowOwner(HWND secondaryWindow, HWND primaryWindow)
+    {
+        // On x64 platform (IntPtr.Size = 8), call SetWindowLongPtr
+        // On x32 platform (IntPtr.Size = 4), call SetWindowLong
+        // Reference: https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowlongptra
+        // Note: CsWin32 will not generate PInvoke.SetWindowLongPtr for x86 platform which will cause a compilation error.
+        if (IntPtr.Size == 8)
+        {
+            SetWindowLongPtr(secondaryWindow, WINDOW_LONG_PTR_INDEX.GWLP_HWNDPARENT, primaryWindow);
+        }
+        else
+        {
+            _ = PInvoke.SetWindowLong(secondaryWindow, WINDOW_LONG_PTR_INDEX.GWL_HWNDPARENT, primaryWindow.Value.ToInt32());
+        }
+    }
+
+    /// <remarks>
+    /// This method cannot be added to the NativeMethods.txt because it is not
+    /// available when compiling the solution for x86 platform.
+    /// </remarks>
+    [DllImport("user32.dll", ExactSpelling = true, EntryPoint = "SetWindowLongPtrW", SetLastError = true)]
+    private static extern nint SetWindowLongPtr(HWND hWnd, WINDOW_LONG_PTR_INDEX nIndex, nint dwNewLong);
+
+    /**********************************************************
+     *  Application services event handlers                   *
+     *  Note: Application services event must be unregistered *
+     *        when the secondary window is closed.            *
+     **********************************************************/
```

Comment:
I guess we should change these to the C# style comments if we're going to keep them